### ISSUE TITLE
Question about "kitty/charsets.c" being compiled twice

### ISSUE THREAD

**Issue by Luflosi** [kovidgoyal/kitty#1570]
I'm rewriting the code that compiles kitty. I noticed, that `kitty/charsets.c` is compiled twice, each time with slightly different command line arguments to the compiler. I need to know if it is sufficient to only compile a file once. The first call to the compiler has these command line options that the second call doesn't have: `-DPRIMARY_VERSION=4000 -DSECONDARY_VERSION=13 -DGL_SILENCE_DEPRECATION -I/usr/local/Cellar/libpng/1.6.37/include/libpng16 -I/usr/local/Cellar/harfbuzz/2.4.0_1/include/harfbuzz -I/usr/local/Cellar/glib/2.60.1/include/glib-2.0 -I/usr/local/Cellar/glib/2.60.1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/graphite2/1.3.13/include`. The second call only has the parameter `-Ikitty`, which the first one doesn't have. Obviously the output file is also different. Would it be sufficient to compile that file only once? Can I use any of the two calls to the compiler?

**Comment by kovidgoyal** [comment]
IIRC charsets.c is used both in kitty and in a kitten, which is why it
is compiled twice. I guess one could technically split it out into a
common module that is imported, but that is way more work than is worth
it just to avoid compiling ~300 lines of code twice.

**Comment by Luflosi** [comment]
The code in my PR currently compiles that file only once and uses the same compiled file for both kitty and the kitten. It seems to be much less work to accomplish this than you thought. Everything compiles just fine, no warnings or error messages are printed during compilation. The CI also didn't find any problems. Does that mean, that everything works as expected or do I need to test if everything works at runtime?

**Comment by kovidgoyal** [comment]
I dont recall, it is used in the choose kitten which has no tests and is
incomplete, so whether it works or not is not easy to determine.

**Comment by Luflosi** [comment]
I'm currently using the compiler flags from the kitten, so the kitten should work fine.

**Comment by kovidgoyal** [comment]
Well, it is a pretty simple file, so I doubt there is any issue with
re-using the object file.
